src/runtime/HalideBuffer.h

```diff
@@ -1381,6 +1396,34 @@ class Buffer {
         std::swap(buf.dim[d1], buf.dim[d2]);
     }
 
+    /** A generalization of transpose: instead of swapping two dimensions,
+     * pass a vector that lists each dimension index exactly once, in the desired order.
+     * For instance, to reorder a 3-dimensional planar image to be interleaved,
+     * pass {2, 0, 1} for storage_order */
+    void reorder(const std::vector<int> &order) {
```

Comment:
I think it would be better to name this "transpose". In Halide::Func, reorder and reorder_storage both have no effect on how something is indexed, but change the underlying representation. Transpose is the complement. It swizzles indexing without changing the underlying representation.